modules/apps/portal-search/portal-search-test/src/testIntegration/java/com/liferay/portal/search/facet/faceted/searcher/test/CategoryFacetTest.java

```diff
@@ -101,43 +109,90 @@ public void testAvoidResidualDataFromDDMStructureLocalServiceTest()
 
 		// See LPS-58543
 
-		String title = "To Do";
+		addCategory("To Do");
+
+		long categoryId = _assetCategory.getCategoryId();
+
+		addUser(_group, categoryId);
+
+		SearchContext searchContext = _getSearchContext(
+			_assetCategory, categoryId);
+
+		Facet facet = _categoryFacetFactory.newInstance(searchContext);
+
+		searchContext.addFacet(facet);
+
+		FacetsAssert.assertFrequencies(
+			facet.getFieldName(), searchContext, search(searchContext),
+			Collections.singletonMap(
+				_getAssetVocabularyCategoryId(_assetCategory), 1));
+	}
+
+	@Test
+	public void testSelection() throws Exception {
+		addCategory(RandomTestUtil.randomString());
 
-		AssetCategory assetCategory = addCategory(title);
+		addJournalArticle(_group, _assetCategory.getTitleCurrentValue());
```

Comment:
you are passing information's from global variables, can you remove the parameters from this method?